### DISCUSSION TITLE
fails on proxy connection but curl succeeds

### DISCUSSION THREAD

**Discussion by KuniyoshiKamimuraCreationline** [cypress-io/cypress#30542] [Questions and Help]
# Summary

My server needs proxy but I can't connect with Cypress.
However, I can connect without Cypress.

## How to reproduce

I ran `cy:e2e` with the following settings in `package.json`, which triggered `cy.visit('/')`, resulting in an error.
```
{
  "scripts": {
    "": "HTTP_PROXY=${HTTP_PROXY_USER}:${HTTP_PROXY_PASSWORD}@***.***.***.***:8080 && cypress open --config baseUrl=https://example.com"
  },
  ...
}
```

```
context('DBM01_01', () => {
  it('retrieve.json', () => {
    cy.visit('/');
    ...
  });
});
```

## What I checked

I confirmed the proxy setting were correctly applied to Cypress by checking the settings screen on Cypress.

![image](https://github.com/user-attachments/assets/36399603-6b5c-4e9f-a5be-669b28a2e568)

# Error details

When I checked the network log, there were no requests to my server.
However Cypress log indicated that `A connection to the upstream proxy could not be established`, suggesting it attempted to connect to my server.

![image](https://github.com/user-attachments/assets/c9d184b9-158b-4417-b8df-bed722f41adb)

## Error message

```
cy.visit() failed trying to load:

https://example.com/

We attempted to make an http request to this URL but the request failed without a response.

We received this error at the network level:

> Error: A connection to the upstream proxy could not be established: 20514496:error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER:....\third_party\boringssl\src\ssl\tls_record.cc:231:
```

## But the same thing with curl succeeds

```
C:\test>curl -i -x %HTTP_PROXY_USER%:%HTTP_PROXY_PASSWORD%@***.***.***.***:8080 https://example.com
HTTP/1.1 200 Connection established

HTTP/1.1 302 Moved Temporarily
```

## Accessing with Chrome without Cypress also succeeds

![image](https://github.com/user-attachments/assets/24be4864-b7c7-4a25-ae0f-240ae9eccaea)

## Accessing with Chrome with Cypress fails

When I manually opened a new tab on chrome with Cypress and entered the URL, it fails with `ERR_EMPTY_RESPONSE`.

![image](https://github.com/user-attachments/assets/8bfc78a2-0009-4db6-b440-5ebf2ececb15)
![image](https://github.com/user-attachments/assets/65d15012-8302-4369-ba8c-83f15229ab93)

# Question

What does `SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER` mean and how can I connect to the proxy?

I've read Cypress' [Proxy Configuration](https://docs.cypress.io/app/references/proxy-configuration) but haven't found the solution.

# My environment

- Windows 11
- Node.js v20.17.0
- Cypress v13.15.0

## Internet settings

My PC uses an auto-configuration script with proxy for my server.

![image](https://github.com/user-attachments/assets/a0496753-8572-412e-8b9f-2a3a3f2edab7)

**Comment by portyu9** [comment]
Hi @KuniyoshiKamimuraCreationline 

The proxy value is the problem.

You currently have a scheme-less value:

user:password@proxy:8080

With Cypress 13.15.0's proxy handling, a missing scheme can be inferred from the destination URL. Since your baseUrl is HTTPS, that can make Cypress treat the proxy itself as HTTPS and attempt a TLS connection to it.

If the proxy on port 8080 is a normal HTTP CONNECT proxy, that explains:

SSL routines: ... WRONG_VERSION_NUMBER

Use an explicit HTTP proxy URL:

HTTP_PROXY=http://user:password@proxy:8080
HTTPS_PROXY=http://user:password@proxy:8080

Using http:// for HTTPS_PROXY is valid: the connection to the proxy is HTTP, and the proxy establishes a CONNECT tunnel for the HTTPS destination.

This also explains why your curl command works: its scheme-less -x proxy is treated as HTTP.

Also note that Cypress does not support PAC files directly, so use the concrete proxy endpoint from your PAC/network configuration. URL-encode any special characters in the username or password.

**Reply by KuniyoshiKamimuraYMSL** [reply]
Thank you.
But I left the project and I can't confirm if it works.
